docs/common/DocsPageTemplate/SideNav/index.vue

```diff
@@ -80,22 +82,62 @@
     watch: {
       filterText(newValue) {
         if (window) {
-          window.sessionStorage.setItem('nav-filter', newValue);
+         //Clear the filter query when filtertext is empty
+        if (!newValue) {
+          this.$router.replace({ path: this.$route.path, query: {} });
         }
-      },
-    },
-    mounted() {
-      if (window) {
-        const filterText = window.sessionStorage.getItem('nav-filter');
-        if (filterText) {
-          this.filterText = filterText;
+        else {
+          //else ,update the filter query param
+          this.$router.push({ path: this.$route.path, query: { ...this.$route.query, filter: newValue}});
+        }
+        this.debouncedUpdateQuery(newValue);
         }
-        this.$refs.links.scrollTop = window.sessionStorage.getItem('nav-scroll');
       }
-      // don't show the nav until the state is set
-      this.loaded = true;
     },
-    methods: {
+   mounted() {
+    if (window) {
+    const { filter } = this.$route.query;
+      // Set filterText from the query parameter if it exists
+    if (filter) {
+      this.filterText = filter;
+    }
+     this.$refs.links.scrollTop = window.sessionStorage.getItem('nav-scroll');
+     // Restoring filter state when a user navigates back
+     window.addEventListener('popstate', (event) => {
+       if (event.state && 'filterText' in event.state) {
+         this.filterText = event.state.filterText;
+       } else {
+         this.filterText = ''; // Reset if no filterText is in state
+       }
+     });
+       // Create debounced function for updating the query
+    this.debouncedUpdateQuery = debounce((newValue) => {
+      if (newValue) {
+        this.$router.push({ path: this.$route.path, query: { ...this.$route.query, filter: newValue } });
+      } 
+      else {
+        this.$router.push({ path: this.$route.path, query: {} });
+      }
+    }, 2000); // 2-second debounce delay
+    }
+
+  // Modify navigation links to preserve the filter query
+  this.$nextTick(() => {
```

Comment:
I would suggest better update [NavLink](https://github.com/AlexVelezLl/kolibri-design-system/blob/e9aa0abdff8c659568cb5dd36a38a3dc847cec2f/docs/common/DocsPageTemplate/SideNav/NavLink.vue) and [DocsLibraryLink](https://github.com/AlexVelezLl/kolibri-design-system/blob/e9aa0abdff8c659568cb5dd36a38a3dc847cec2f/docs/common/DocsPageTemplate/SideNav/NavLink.vue) to pass the query params, so we have dont have to add an event listener to all links of the page :)